Framework/CurveFitting/inc/MantidCurveFitting/FortranMatrix.h

```diff
@@ -0,0 +1,166 @@
+#ifndef MANTID_CURVEFITTING_FORTRANMATRIX_H_
+#define MANTID_CURVEFITTING_FORTRANMATRIX_H_
+
+#include <cstddef>
+#include <stdexcept>
+
+namespace Mantid {
+namespace CurveFitting {
+
+/** FortranMatrix is a wrapper template for GSLMatrix  and ComplexMatrix
+  to simplify porting fortran programs to C++.
+  This matrix allows to use arbitrary index bases as they do in
+  fortran. Indexing can begin with any integer number including
+  negative.
+
+  Copyright &copy; 2016 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
+  National Laboratory & European Spallation Source
+
+  This file is part of Mantid.
+
+  Mantid is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 3 of the License, or
+  (at your option) any later version.
+
+  Mantid is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+  File change history is stored at: <https://github.com/mantidproject/mantid>
+  Code Documentation is available at: <http://doxygen.mantidproject.org>
+*/
+template <class MatrixClass> class FortranMatrix : public MatrixClass {
+  /// Base for the first index
+  int m_base1;
+  /// Base for the second index
+  int m_base2;
+  /// Typedef the types returned by the base class's operators []. They aren't
+  /// necessarily the same as the stored type (double or complex).
+  typedef decltype(
+      std::declval<const MatrixClass>().operator()(0, 0)) ElementConstType;
+  typedef decltype(std::declval<MatrixClass>().operator()(0, 0)) ElementRefType;
+
+public:
+  /// Constructor
+  FortranMatrix();
+  /// Constructor
+  FortranMatrix(const int nx, const int ny);
+  /// Copy constructor
+  FortranMatrix(const FortranMatrix &M);
```

Comment:
Does this need to be defined. It looks like a member-wise copy which the default would do. By defining this here it suppresses the move operations..